snips_nlu/intent_parser/probabilistic_intent_parser.py

```diff
@@ -55,16 +62,23 @@ def fit(self, dataset, force_retrain=True):
         Returns:
             :class:`ProbabilisticIntentParser`: The same instance, trained
         """
+        logger.info("Fitting probabilistic intent parser...")
         dataset = validate_and_format_dataset(dataset)
         intents = list(dataset[INTENTS])
         if self.intent_classifier is None:
             self.intent_classifier = build_processing_unit(
                 self.config.intent_classifier_config)
         if force_retrain or not self.intent_classifier.fitted:
+            logger.debug("Fitting intent classifier...")
+            intent_classifier_start = datetime.now()
             self.intent_classifier.fit(dataset)
+            logger.debug("Fitted intent classifier in %s",
+                         elapsed_since(intent_classifier_start))
```

Comment:
Can't you move this into the `fit` method of the intent classifier and use the decorator?